bin/makePsfLibrary.py

```diff
@@ -0,0 +1,175 @@
+                ds = afwDetection.FootprintSet(subImage, afwDetection.Threshold(1000))
+                sources = afwTable.SourceCatalog(schema)
+                record = sources.addNew()
+                record.setFootprint(ds.getFootprints()[0])
+                plugin.measure(record, exp)
+        
+                # shift the bounding box to the correct centroid
+                newCoord = exp.getWcs().pixelToSky(afwGeom.Point2D(record.get("_x"), record.get("_y")))
+                dx = int(round(record.get("_x"))) - x
+                dy = int(round(record.get("_y"))) - y
+                starBBox.shift(afwGeom.Extent2I(dx, dy))
+        
+                # if the exposure still contains the cutout, use it.
+                if exp.getBBox().contains(starBBox):
+                    subImage = afwImage.ImageF(exp.getMaskedImage().getImage(), starBBox)
+        
+                    # convert the image to 64 bits, and turn it into a Psf
+                    # then warp it using the xytransform from above
+                    data = subImage.getArray().astype(numpy.float64)
+                    kernel = afwMath.FixedKernel(afwImage.ImageD(data))
+                    psf = measAlg.KernelPsf(kernel)
+                    warpedPsf = measAlg.WarpedPsf(psf, xytransform, "lanczos3");
+                    if len(outputHdus) == 0:
+                        hdu = pyfits.hdu.image.PrimaryHDU()
+                    else:
+                        hdu = pyfits.hdu.image.ImageHDU()
+                    hdu.data = warpedPsf.computeImage().getArray()
+                    hdu.header.append("GS_SCALE")
+                    hdu.header.set("GS_SCALE", .2)
+                    hdu.header.append("PSF_NO")
+                    hdu.header.set("PSF_NO",psfIndex)
+                    
+                    outputHdus.append(hdu)
+                    fout.write("%s %d %d\n"%(os.path.basename(output_file), psfCount, psfIndex))
+                    psfCount = psfCount + 1
+            psfIndex = psfIndex + 1
+        cat.close()
+        del hdu
+        del exp
+    outputHdus.writeto(output_file, clobber=True)
+    print psfCount, " psf images were cutout from the original image", len(outputHdus)
+
+#   This main program creates a library for each raft in the input_directory.  A raft is assumed
+#   to be comprised of 9 files named basename_Rnn_Smm.fits in the input directory
+#   The cutouts from each set of 9 files is placed in the output_dir in basename_Rnn.fits
+#   An index of all the libraries created in this way is created in basename.index
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("catalog", help="Name of the star catalog fed to OpSim")
+    parser.add_argument("input_dir", help="Name of the star grid produced by OpSim")
+    parser.add_argument("basename", help="Name of the star grid produced by OpSim")
+    parser.add_argument("output_dir", help="Name of the output directory")
```

Comment:
It would be nice if these help messages contained the same information as the docstring towards the start of the file.